modules/loader/src/main/java/com/opengamma/strata/loader/impl/fpml/CdsFpmlParserPlugin.java

```diff
@@ -194,6 +197,17 @@ Trade parseCds(FpmlDocument document, XmlElement tradeEl, TradeInfoBuilder trade
     throw new FpmlParseException("FpML CDS must be single name or index");
   }
 
+  private Optional<StandardId> tryParseIndexId(XmlElement indexInformation) {
+    Optional<XmlElement> idOpt = indexInformation.findChild("indexId");
+    if (idOpt.isPresent()) {
+      XmlElement idElement = idOpt.get();
+      String id = idElement.getContent();
+      String indexIdScheme = idElement.findAttribute("indexIdScheme").orElse(StandardSchemes.RED9_SCHEME);
```

Comment:
I think this logic makes more sense:
* Is it length 9, then set as RED9
* Is the indexIdScheme attribute set, then use it
* Else return `Optional.empty`